crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -197,11 +197,11 @@ impl<'src> Parser<'src> {
                 let start = self.node_start();
                 let parsed_expr = self.parse_star_expression_list();
 
-                if self.eat(TokenKind::Equal) {
+                if self.at(TokenKind::Equal) {
                     Stmt::Assign(self.parse_assign_statement(parsed_expr, start))
-                } else if self.eat(TokenKind::Colon) {
+                } else if self.at(TokenKind::Colon) {
```

Comment:
I think it makes more sense that the parsing function starts with the expected token. It's also consistent with how other parsing function works.